api/base/serializers.py

```diff
@@ -279,6 +299,53 @@ def to_representation(self, data):
             self.child.to_representation(item, envelope=None) for item in data
         ]
 
+    # Overrides ListSerializer which doesn't support multiple update by default
+    def update(self, instance, validated_data):
+        if len(instance) != len(validated_data):
+            raise exceptions.NotFound()
+        instance_mapping = {item._id: item for item in instance}
```

Comment:
https://github.com/CenterForOpenScience/osf.io/pull/4353/files#diff-4b396d38a0947821679352d308445c5aR306  For when _id is not the public-facing identifier I've allowed an optional field on Meta, unique_identifier.